x-pack/plugins/fleet/server/plugin.ts

```diff
@@ -416,8 +417,29 @@ export class FleetPlugin
           summary: 'Fleet is setting up',
         });
 
+        await plugins.licensing.license$.pipe(take(1)).toPromise();
+
         await setupFleet(
-          new SavedObjectsClient(core.savedObjects.createInternalRepository()),
+          core.savedObjects.getScopedClient(
```

Comment:
Wouldn't we need to do this not only on this API call but on any call to the outputs service that needs to write or read output credentials? I think it might be better to have the output service use it's own SO client rather than using the one that is passed in. Something like this maybe?

```ts
class OutputService {
  constructor() {
    this.encryptedSoClient = appContextService.getSavedObjects().getScopedClient(
	      // Use a fake request to get a client with encrypted saved object
	      {
	        headers: {},
	        getBasePath: () => '',
	        path: '/',
	        route: { settings: {} },
	        url: {
	          href: '/',
	        },
	        raw: {
	          req: {
	            url: '/',
	          },
	        },
	      },
	      {
	        excludedWrappers: ['security', 'spaces'],
	      }
    );
  }

  // in each method
  public create(...) {
    this.encryptedSoClient.create(...)
  }
}
```

Reply:
Yes it will probably a safer option 👍